chainer/links/connection/tree_lstm.py

```diff
@@ -0,0 +1,224 @@
+from chainer.functions.activation import tree_lstm
+from chainer.functions.array import concat
+from chainer.functions.array import split_axis
+from chainer.links.connection import linear
+
+from chainer import cuda
+from chainer import link
+from chainer import utils
+from chainer import variable
+
+
+class ChildSumTreeLSTM(link.Chain):
+    """Child-Sum TreeLSTM unit.
+
+    This is a Child-Sum TreeLSTM unit as a chain.
+    This link is a variable arguments function, which compounds
+    the states of all children nodes into the new states of
+    a current (parent) node. *states* denotes the cell state, c,
+    and the output, h, which are produced by this link.
+    This link doesn't keep cell and hidden states internally.
+
+    For example, this link is called such as
+    ``func(c1, c2, h1, h2, x)`` if the number of children nodes is 2,
+    while ``func(c1, c2, c3, h1, h2, h3, x)`` if that is 3.
+    This function is *independent* from an order of children nodes.
+    Thus, the returns of ``func(c1, c2, h1, h2, x)`` equal to
+    those of ``func(c2, c1, h2, h1, x)``.
+
+    Args:
+        in_size (int): Dimension of input vectors.
+        out_size (int): Dimensionality of cell and output vectors.
+
+    Attributes:
+        W_x (chainer.links.Linear): Linear layer of
+            connections from input vectors.
+        W_h_aio (chainer.links.Linear): Linear layer of connections between
+            (:math:`a`, :math:`i`, :math:`o`) and summation of children's
+            output vectors. :math:`a`, :math:`i` and :math:`o` denotes
+            input compound,
+            input gate and output gate, respectively.
+            :math:`a`, input compound, equals to :math:`u` in
+            the paper by Tai et al.
+        W_h_f (chainer.links.Linear): Linear layer of connections between
+            forget gate :math:`f` and the output of each child.
+
+    See the paper for details: `Improved Semantic Representations From \
+    Tree-Structured Long Short-Term Memory Networks \
+    <http://www.aclweb.org/anthology/P15-1150>`_.
+
+    """
+
+    def __init__(self, in_size, out_size):
+        super(ChildSumTreeLSTM, self).__init__(
+            W_x=linear.Linear(in_size, 4 * out_size),
+            W_h_aio=linear.Linear(out_size, 3 * out_size, nobias=True),
+            W_h_f=linear.Linear(out_size, out_size, nobias=True),
+        )
+        self.in_size = in_size
+        self.state_size = out_size
+        utils.experimental('chainer.links.tree_lstm.py')
+
+    def __call__(self, *cshsx):
```

Comment:
This also follows the reason above for the functions and `chainer.links.StatelessLSTM.__call__(c, h, x)`.